### ISSUE TITLE
Training error : terminate called after throwing an instance of 'std::invalid_argument'   what():  stod

### ISSUE THREAD

**Issue by samin9796** [facebookresearch/wav2letter#424]
This is what I am getting after I run the train command.

(wav2letter) shafkat@server-master:/data/ahnaf/wav2letter/wav2letter/build$ ./Train train --flagsfile=/data/ahnaf/wav2letter/dataset_prep/flagfile.cfg --logtostderr=1 --reportiters=1
I1011 02:19:08.658896 25683 Train.cpp:58] Reading flags from file /data/ahnaf/wav2letter/dataset_prep/flagfile.cfg
I1011 02:19:09.670284 25683 Train.cpp:137] Gflags after parsing 
--flagfile=; --fromenv=; --tryfromenv=; --undefok=; --tab_completion_columns=80; --tab_completion_word=; --help=false; --helpfull=false; --helpmatch=; --helpon=; --helppackage=false; --helpshort=false; --helpxml=false; --version=false; --adambeta1=0.90000000000000002; --adambeta2=0.999; --am=; --arch=network.arch; --archdir=/data/ahnaf/wav2letter/dataset_prep/; --attention=content; --attentionthreshold=0; --attnWindow=no; --attnconvchannel=0; --attnconvkernel=0; --attndim=0; --batchsize=4; --beamsize=2500; --beamthreshold=25; --blobdata=false; --channels=1; --criterion=asg; --critoptim=sgd; --datadir=/data/ahnaf/wav2letter/dataset_prep/; --dataorder=input; --decoderattnround=1; --decoderdropout=0; --decoderrnnlayer=1; --decodertype=wrd; --devwin=0; --emission_dir=; --enable_distributed=false; --encoderdim=0; --eostoken=false; --everstoredb=false; --fftcachesize=1; --filterbanks=40; --flagsfile=/data/ahnaf/wav2letter/dataset_prep/flagfile.cfg; --framesizems=25; --framestridems=10; --gamma=1; --gumbeltemperature=1; --hardselection=1; --input=wav; --inputbinsize=100; --inputfeeding=false; --iter=3; --itersave=false; --labelsmooth=0; --leftWindowSize=50; --lexicon=/data/ahnaf/wav2letter/dataset_prep/lexicon.txt; --linlr=-1; --linlrcrit=-1; --linseg=0; --lm=; --lm_memory=5000; --lm_vocab=; --lmtype=kenlm; --lmweight=0; --localnrmlleftctx=0; --localnrmlrightctx=0; --logadd=false; --lr=0.0001; --lrcosine=false; --lrcrit=0.0001; --maxdecoderoutputlen=200; --maxgradnorm=0; --maxisz=9223372036854775807; --maxload=-1; --maxrate=10; --maxsil=50; --maxtsz=9223372036854775807; --maxword=-1; --melfloor=1; --memstepsize=10485760; --mfcc=false; --mfcccoeffs=13; --mfsc=true; --minisz=0; --minrate=3; --minsil=0; --mintsz=0; --momentum=0; --netoptim=sgd; --noresample=false; --nthread=1; --nthread_decoder=1; --numattnhead=8; --onorm=none; --optimepsilon=1e-08; --optimrho=0.90000000000000002; --outputbinsize=5; --pctteacherforcing=100; --pcttraineval=100; --pow=false; --pretrainWindow=0; --replabel=0; --reportiters=1; --rightWindowSize=50; --rndv_filepath=; --rundir=/data/ahnaf/wav2letter/dataset_prep/; --runname=first; --samplerate=16000; --sampletarget=0; --samplingstrategy=rand; --sclite=; --seed=0; --show=false; --showletters=false; --silweight=0; --smearing=none; --smoothingtemperature=1; --softselection=inf; --softwoffset=10; --softwrate=5; --softwstd=5; --sqnorm=false; --stepsize=1000000; --surround=; --tag=; --target=tkn; --test=; --tokens=tokens.txt; --tokensdir=/data/ahnaf/wav2letter/dataset_prep/; --train=train.lst; --trainWithWindow=false; --transdiag=0; --unkweight=-inf; --uselexicon=true; --usewordpiece=false; --valid=valid.lst; --weightdecay=0; --wordscore=0; --wordseparator=|; --world_rank=0; --world_size=1; --alsologtoemail=; --alsologtostderr=false; --colorlogtostderr=false; --drop_log_memory=true; --log_backtrace_at=; --log_dir=; --log_link=; --log_prefix=true; --logbuflevel=0; --logbufsecs=30; --logemaillevel=999; --logfile_mode=436; --logmailer=/bin/mail; --logtostderr=true; --max_log_size=1800; --minloglevel=0; --stderrthreshold=2; --stop_logging_if_full_disk=false; --symbolize_stacktrace=true; --v=0; --vmodule=; 
I1011 02:19:09.670956 25683 Train.cpp:138] Experiment path: /data/ahnaf/wav2letter/dataset_prep/first
I1011 02:19:09.670964 25683 Train.cpp:139] Experiment runidx: 1
I1011 02:19:09.671581 25683 Train.cpp:183] Number of classes (network): 74
I1011 02:19:09.731966 25683 Train.cpp:190] Number of words: 30681
I1011 02:19:09.752733 25683 Train.cpp:204] Loading architecture file from /data/ahnaf/wav2letter/dataset_prep/network.arch
I1011 02:19:10.517838 25683 Train.cpp:225] [Network] Sequential [input -> (0) -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> output]
	(0): View (-1 1 40 0)
	(1): Conv2D (40->300, 48x1, 2,1, SAME,SAME, 1, 1) (with bias)
	(2): ReLU
	(3): Conv2D (300->300, 32x1, 1,1, 0,0, 1, 1) (with bias)
	(4): ReLU
	(5): Reorder (2,0,3,1)
	(6): Linear (300->74) (with bias)
I1011 02:19:10.517908 25683 Train.cpp:226] [Network Params: 3478874]
I1011 02:19:10.517916 25683 Train.cpp:227] [Criterion] AutoSegmentationCriterion
I1011 02:19:10.517926 25683 Train.cpp:235] [Network Optimizer] SGD
I1011 02:19:10.517935 25683 Train.cpp:236] [Criterion Optimizer] SGD
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stod
*** Aborted at 1570785550 (unix time) try "date -d @1570785550" if you are using GNU date ***
PC: @     0x7ff352233e97 gsignal
*** SIGABRT (@0x3eb00006453) received by PID 25683 (TID 0x7ff379b8e380) from PID 25683; stack trace: ***
    @     0x7ff378a8c890 (unknown)
    @     0x7ff352233e97 gsignal
    @     0x7ff352235801 abort
    @     0x7ff352e57957 (unknown)
    @     0x7ff352e5dab6 (unknown)
    @     0x7ff352e5daf1 std::terminate()
    @     0x7ff352e5dd24 __cxa_throw
    @     0x7ff352e597e3 (unknown)
    @     0x56128009a7ac w2l::W2lListFilesDataset::loadListFile()
    @     0x56128009b38d w2l::W2lListFilesDataset::W2lListFilesDataset()
    @     0x5612800b526e w2l::createDataset()
    @     0x56127fea5601 main
    @     0x7ff352216b97 __libc_start_main
    @     0x56127fef870a _start
Aborted (core dumped)
(wav2letter) shafkat@server-master:/data/ahnaf/wav2letter/wav2letter/build$

**Comment by lunixbochs** [comment]
This is a gflags bug I think, if your locale separates floating point numbers with commas or something instead of decimals it will break all wav2letter models.

Try this:
```
export LC_NUMERIC="en_US.UTF-8"
```

But don't close this issue even if that works, as wav2letter should use a parser that doesn't break in this way.

**Comment by samin9796** [comment]
There were spaces in the wav files name. I renamed all the wav files and fixed the problem.
